### PR TITLE
CODENVY-174 pre-select correct branch when start point is set

### PR DESCRIPTION
**Pull request by mshaposhnik**
Signed-off-by: Max Shaposhnik mshaposhnik@codenvy.com

### CODE REVIEW

**Comment by mshaposhnik** [comment]
@skabashnyuk @sleshchenko @akorneta

File: core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java [hunk 1]
```diff
@@ -45,17 +47,17 @@ public Void execute() throws GitException {
...
-	            commandLine.add("-b");
-	            commandLine.add(branchName);
-	        } else if (branchName != null) {
-	            commandLine.add(branchName);
+            if (createNew) {
+                commandLine.add("-b");
+                commandLine.add(branchName);
+            } else if (branchName != null) {
+                commandLine.add(branchName);
 	        }
```
Comment (sleshchenko) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java]: check formatting here

File: core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/shared/CheckoutRequest.java [hunk 2]
```diff
@@ -60,4 +60,10 @@
     
     CheckoutRequest withFiles(List<String> files);
 
+    void setNoTrack(boolean noTrack);
+
+    boolean isNoTrack();
```
Comment (sleshchenko) [inline core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/shared/CheckoutRequest.java:66]: Please add java docs for this parameter

File: core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java [hunk 3]
```diff
@@ -23,9 +23,11 @@
 public class CheckoutCommand extends GitCommand<Void> {
 
     private boolean createNew;
-    private String  branchName;
-    private String  trackBranch;
-    private String  startPoint;
+
```
Comment (sleshchenko) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java]: remove empty line

File: core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java (hunk 1 above)
Comment (mshaposhnik) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java]: it is ok here (dont trust your eyes, hi)

**Comment by sleshchenko** [comment]
other LGTM

File: core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java [hunk 4]
```diff
@@ -45,18 +46,18 @@ public Void execute() throws GitException {
...
-	            commandLine.add(branchName);
-	        }
+            if (createNew) {
+                commandLine.add("-b");
+                commandLine.add(branchName);
+            } else if (branchName != null) {
+                commandLine.add(branchName);
+            }
 	
 	        if (trackBranch != null) {
```
Comment (sleshchenko) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java]: check formatting here :smiley:

Comment (mshaposhnik) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/CheckoutCommand.java]: there is tabs at line beginnig, i cant see them

**Comment by akorneta** [comment]
ok

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**